src/main/java/com/onarandombox/MultiverseCore/listeners/MVPlayerListener.java

```diff
@@ -253,112 +243,140 @@ public void playerPortalCheck(PlayerPortalEvent event) {
             return;
         }
     }
+
     /**
      * This method is called when a player actually portals via a vanilla style portal.
      * @param event The Event that was fired.
      */
     @EventHandler(priority = EventPriority.HIGH)
     public void playerPortal(PlayerPortalEvent event) {
-        if (event.isCancelled() || (event.getFrom() == null)) {
-            return;
-        }
-        // The adjust should have happened much earlier.
-        if (event.getTo() == null) {
+        if (event.isCancelled() || event.getFrom() == null || event.getTo() == null) {
             return;
         }
-        MVWorld fromWorld = this.worldManager.getMVWorld(event.getFrom().getWorld().getName());
-        MVWorld toWorld = this.worldManager.getMVWorld(event.getTo().getWorld().getName());
-        if (event.getFrom().getWorld().equals(event.getTo().getWorld())) {
-            // The player is Portaling to the same world.
-            Logging.finer("Player '" + event.getPlayer().getName() + "' is portaling to the same world.");
-            return;
-        }
-        event.setCancelled(!pt.playerHasMoneyToEnter(fromWorld, toWorld, event.getPlayer(), event.getPlayer(), true));
-        if (event.isCancelled()) {
-            Logging.fine("Player '" + event.getPlayer().getName()
-                    + "' was DENIED ACCESS to '" + event.getTo().getWorld().getName()
-                    + "' because they don't have the FUNDS required to enter.");
+
+        MVWorld fromWorld = this.worldManager.getMVWorld(event.getFrom().getWorld());
+        MVWorld toWorld = this.worldManager.getMVWorld(event.getTo().getWorld());
+        ActionResponse actionCheckResponse = this.actionChecker.canGoFromToWorld(event.getPlayer(), event.getPlayer(), fromWorld, toWorld);
+        Logging.fine(actionCheckResponse.toString());
+        if (!actionCheckResponse.isSuccessful()) {
```

Comment:
Just thought about this here... I think we should generally be looking to avoid negated logic. We can do this by adding a `actionCheckResponse.isFailure`.